core/src/components/text/text.ionic.scss

```diff
@@ -0,0 +1,247 @@
+// --------------------------------------------------
+/// Font Family
+$os-font-family: "Inter", sans-serif;
+
+/// Font sizes
+/// @prop - Font size small
+$os-font-size-s: 12px;
+
+/// @prop - Font size base
+$os-font-size-base: 14px;
+
+/// @prop - Font size large
+$os-font-size-l: 16px;
+
+/// Font size displays
+/// @prop - Display font size small
+$os-display-s: 32px;
+
+/// @prop - Display font size medium
+$os-display-m: 36px;
+
+/// Headings
+/// @prop - Font size of heading level 1
+$os-h1-font-size: dynamic-font(28px) !default;
+
+/// @prop - Font size of heading level 2
+$os-h2-font-size: dynamic-font(26px) !default;
+
+/// @prop - Font size of heading level 3
+$os-h3-font-size: dynamic-font(24px) !default;
+
+/// @prop - Font size of heading level 4
+$os-h4-font-size: dynamic-font(22px) !default;
+
+/// @prop - Font size of heading level 5
+$os-h5-font-size: dynamic-font(20px) !default;
+
+/// @prop - Font size of heading level 6
+$os-h6-font-size: dynamic-font(18px) !default;
+
+/// Font Weights
+/// @prop - Font weight light
+$os-font-light: 300;
+
+/// @prop - Font weight regular
+$os-font-regular: 400;
+
+/// @prop - Font weight medium
+$os-font-medium: 500;
+
+/// @prop - Font weight semibold
+$os-font-semibold: 600;
+
+/// @prop - Font weight bold
+$os-font-bold: 700;
+
+/// Line heights
+/// @prop - Line height size xsmall
+$os-line-height-xs: 16px;
+
+/// @prop - Line height size small
+$os-line-height-s: 20px;
+
+/// @prop - Line height size medium
+$os-line-height-m: 24px;
+
+/// @prop - Line height size large
+$os-line-height-l: 28px;
+
+/// @prop - Line height size xlarge
+$os-line-height-xl: 32px;
+
+/// @prop - Line height size xxlarge
+$os-line-height-xxl: 36px;
+
+/// @prop - Display line height small
+$os-line-height-display-s: 44px;
+
+/// @prop - Display line height medium
+$os-line-height-display-m: 48px;
+
+/// Letter spacing
+/// @prop - Letter spacing none
+$os-letter-spacing-none: 0em;
+
+/// @prop - Letter spacing small
+$os-letter-spacing-s: 0.1em;
+
+/// @prop - Letter spacing medium
+$os-letter-spacing-m: 0.15em;
+
+html {
+  font-family: $os-font-family;
+  font-size: $os-font-size-base;
+  font-weight: $os-font-regular;
+  line-height: $os-line-height-m;
+}
+
+h1,
+h2,
+h3,
+h4,
+h5,
+h6 {
+  font-weight: $os-font-semibold;
+}
+
+h1 {
+  font-size: $os-h1-font-size;
+  line-height: $os-line-height-xxl;
+}
+
+h2 {
+  font-size: $os-h2-font-size;
+  line-height: $os-line-height-xl;
+}
+
+h3 {
+  font-size: $os-h3-font-size;
+  line-height: $os-line-height-l;
+}
+
+h4 {
+  font-size: $os-h4-font-size;
+  line-height: $os-line-height-l;
+}
+
+h5 {
+  font-size: $os-h5-font-size;
+  line-height: $os-line-height-l;
+}
+
+h6 {
+  font-size: $os-h6-font-size;
+  line-height: $os-line-height-s;
+}
+
+/// Utility Classes
+/// Headings
+.os-heading1,
+.os-heading2,
+.os-heading3,
+.os-heading4,
+.os-heading5,
+.os-heading6 {
+  font-weight: $os-font-semibold;
+}
+
+.os-heading1 {
+  font-size: $os-heading1;
```

Comment:
Many of these Sass variables are undefined and fail to compile in the Stencil project. Are there additional diffs that are local to your environment?

Reply:
This approach was reviewed to work with Sass variables and CSS variables.